tests/drivers/display/display_read_write/src/main.c

```diff
@@ -0,0 +1,268 @@
+			} else {
+				zassert_equal(buf8[i], (uint8_t)color, "@%d", i);
+			}
+			break;
+		}
+	}
+}
+
+/**
+ * Fill the buffer with 0 before running tests.
+ */
+static void display_before(void *text_fixture)
+{
+	display_get_capabilities(dev, &cfg);
+	bpp = bytes_per_pixel(cfg.current_pixel_format);
+	is_tiled = ((bpp == 1) && (cfg.screen_info & SCREEN_INFO_MONO_VTILED));
+
+	struct display_buffer_descriptor desc = {
+		.height = display_height,
+		.pitch = display_width,
+		.width = display_width,
+		.buf_size = display_height * display_width * bpp,
+	};
+
+	memset(disp_buffer, 0, sizeof(disp_buffer));
+	display_write(dev, 0, 0, &desc, disp_buffer);
+}
+
+/*
+ * Verify that we can get a color of '0' from all pixels
+ * when after zeroing the buffer.
+ */
+ZTEST(display_read_write, test_clear)
+{
+	verify_background_color(0, 0, display_width, display_height, 0);
+}
+
+/*
+ * Write to the head of the buffer and check that
+ * the same value can be read.
+ */
+ZTEST(display_read_write, test_write_to_buffer_head)
+{
+	uint8_t data[4] = {0xFA, 0xAF, 0x9F, 0xFA};
+	uint8_t height = (is_tiled ? 8 : 1);
+	uint16_t width = sizeof(data) / bpp;
+	uint16_t buf_size = width * bpp;
+	struct display_buffer_descriptor desc = {
+		.height = height,
+		.pitch = width,
+		.width = width,
+		.buf_size = buf_size,
+	};
+
+	/* write data to head of buffer */
+	display_write(dev, 0, 0, &desc, data);
+
+	/* check write data and read data are same */
+	verify_bytes_of_area(data, 0, 0, width, height);
+
+	/* check remaining region still black */
+	verify_background_color(0, height, display_width, display_height - height, 0);
+	verify_background_color(width, 0, display_width - width, display_height, 0);
+}
+
+/*
+ * Write to the tail of the buffer and check that
+ * the same value can be read.
+ */
+ZTEST(display_read_write, test_write_to_buffer_tail)
+{
+	uint8_t data[4] = {0xFA, 0xAF, 0x9F, 0xFA};
+	uint16_t height = (is_tiled ? 8 : 1);
+	uint16_t width = sizeof(data) / bpp;
+	uint16_t buf_size = width * bpp;
+	struct display_buffer_descriptor desc = {
+		.height = height,
+		.pitch = width,
+		.width = width,
+		.buf_size = buf_size,
+	};
+	struct display_buffer_descriptor desc_whole = {
+		.height = display_height,
+		.pitch = display_width,
+		.width = display_width,
+		.buf_size = display_height * display_width * bpp / height,
+	};
+	int err;
+
+	/* write data to tail of buffer */
+	display_write(dev, display_width - width, display_height - height, &desc, data);
+
+	/* read entire displayed data */
+	err = display_read(dev, 0, 0, &desc_whole, disp_buffer);
+	zassert_ok(err, "display_read failed");
+
+	/* check write data and read data are same */
+	if (is_tiled) {
+		zassert_mem_equal(data,
+				  disp_buffer + (display_width * display_height / 8 - buf_size),
+				  buf_size, NULL);
+	} else {
+		zassert_mem_equal(data,
+				  disp_buffer + (display_width * display_height * bpp - buf_size),
+				  buf_size, NULL);
+	}
+
+	/* check remaining region still black */
```

Comment:
Nit:
`/* check remaining region still black */`
->
`/* Check remaining region is still black */`

Reply:
Fixed it.